daffodil-cli/src/it/resources/org/apache/daffodil/CLI/input/gen_blob.py

```diff
@@ -0,0 +1,18 @@
+#! /usr/bin/python
+
+import argparse
+import binascii
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-s", "--size", help="Specify the size of the generated file in MB", type=int)
+    parser.add_argument("-o", "--output", help="Path to the file that will be generated")
+    args = parser.parse_args()
+
+    bytes_written = 0
+    with open(args.output, 'wb') as fout:
+        while (bytes_written < (args.size * 1024000)):
+            fout.write(binascii.unhexlify('deadbeef'))
+            bytes_written += 4
+
+    print("Wrote %d bytes" % bytes_written)
```

Comment:
Was this used to generate 2_01GB.zip? In general, Apache reccommends that anything that is generated not be included in a repo, but instead include instructions on how to generate it. Either way a user needs to perform an action to get the 2GB file (either run this script or unzip a file), so I'd rather them just run this script.

Another though, it might make for a more interesting test if the first 8 bytes were an unsigned long of the number of bytes. That's more realistic to how most blob files are going to look, and it allows the schema to have outputValueCalc for unparse and runtime calculated lengths parse. And then you can have a single schema that could parse anything that this scrip could generate without having to modify the schema.

Reply:
I'm a little confused about the schema having an outputValueCalc. I updated the schema to parse the first 8 bytes as blobLength, and then had the actual blob element reference that length. Should I be using an outputValueCalc somewhere?

Reply:
Yes, have that length Element have an OVC: like this:
`dfdl:outputValueCalc="{ dfdl:valueLength(../blobelement) }"`